server/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { Body, Controller, Get, Post, Req, Res, Session, UseGuards } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { AuthGuard } from '@nestjs/passport';
+import { Request, Response } from 'express';
+import { AppConfigService } from 'src/config/app.service';
+import { UserService } from 'src/models/users/user.service';
+import { AuthService } from './auth.service';
+import { LoginDto } from './dtos/login.dto';
+import { JwtAuthGuard } from './jwt-auth.guard';
+
+@Controller('/api/v1/auth')
+export class AuthController {
+  private clientUrl: string;
+  constructor(
+    private authService: AuthService,
+    private userService: UserService,
+    private jwtService: JwtService,
+    appConfigService: AppConfigService,
+  ) {
+    this.clientUrl = appConfigService.client;
+  }
+
+  @Get('/')
+  requestAuth(@Res() res: Response) {
+    res.redirect(this.authService.getRequestUrl());
+  }
+
+  @Get('redirect')
+  async requestRedirect(@Session() session: Record<string, any>, @Req() req: Request, @Res() res: Response) {
+    const { code, state } = req.query as Record<string, string>;
+    console.log(code, state);
+    const { oAuthId, email } = await this.authService.oauth(code, state);
+
+    // 가입한 유저라면 jwt 발급하고 메인으로 redirect
+    const user = await this.userService.findByOauthId(oAuthId);
+    if (user) {
+      res.cookie('jwt', this.jwtService.sign({ userId: user.id }));
+      res.redirect(this.clientUrl);
+    }
+
+    res.cookie('jwt', this.jwtService.sign({ userId: '123' }));
+
+    // 가입하지 않은 유저는 session에 oAuthId 담고 약관동의로 redirect
+    session.oAuthId = oAuthId;
+    res.redirect(`${this.clientUrl}/signup?email=${email}`);
+  }
+
+  @UseGuards(JwtAuthGuard)
+  @Get('/user')
+  async getUser(@Req() req: Request) {
+    const { id, email } = req.cookies;
+    return { id, email };
+  }
+
+  @UseGuards(AuthGuard('local'))
+  @Post('/login')
+  async login(@Req() req: Request, @Res() res: Response) {
+    res.cookie('jwt', this.authService.login(req.user));
+  }
+
+  @Get('/logout')
+  async logout(@Req() req: Request, @Res() res: Response) {
+    res.clearCookie('jwt');
+    req.session.destroy(() => {});
+    return { ok: true };
+  }
```

Comment:
```suggestion
  @Get('/logout')
  async logout(@Req() req: Request, @Res({ passthrough: true }) res: Response) {
    res.clearCookie('jwt');
    req.session.destroy();
    return { ok: true };
  }
```

`@Res({ passthrough: true }) res: Response`로 해줘야 정상적으로 return이 되는것으로 알고 있습니다.

`destroy()`는 callback이 없어도 되지 않나요?

Reply:
맞아요 res.send 로 직접 명시하거나 passthrough 줘야되는데 빼먹었네요. 고쳤습니다
session destory 는 콜백필요합니다! 
![스크린샷 2021-08-24 오후 9 40 26](https://user-images.githubusercontent.com/50898502/130617943-1e83863a-b437-40f8-ac43-fe033082acdf.png)